esm_runscripts/namelists.py

```diff
@@ -126,7 +126,33 @@ def nmls_remove(mconfig):
                     value = change_entries[key]
                     if value == "remove_from_namelist":
                         namelist_removes.append((namelist, change_chapter, key))
+
+                        # the key is probably coming from esm_tools config
+                        # files or from a user runscript. It can contain lower
+                        # case, but the original Fortran namelist could be in
+                        # any case combination. Here `original_key` is coming
+                        # from the default namelist and may contain mixed case.
+                        # `key` is the processed variable my f90nml package and
+                        # is lowercase
+                        remove_original_key = False
+
+                        # traverse the namelist chapter and see if a mixed case
+                        # variable is also found
+                        for key2 in namelist_changes[namelist][change_chapter]:
+                            # take care of the MiXeD FORTRAN CaSeS
+                            if key2.lower() == key.lower() and key2 != key:
+                                original_key = key2
+                                remove_original_key = True
+                                namelist_removes.append((namelist, change_chapter, original_key))
```

Comment:
At some point (in the distant future), I'd like to overhaul this and name it a namedtuple or something. While it makes sense, it'd be nice to have some sort of clean way of sending what we actually are removing. Right now, it still needs to be explicitly in the correct order, that might induce confusion at some point.